eval/udscore.py

```diff
@@ -81,44 +78,51 @@
         ),
     )
     print("==", filedata, "==")
-    for tokenlist in parse_incr(data_file):
-        for token in tokenlist:
-            error_flag = False
-            if token["lemma"] == "_":  # or token['upos'] in ('PUNCT', 'SYM')
-                # flag = True
-                continue
+    with open(filename, "r", encoding="utf-8") as data_file:
+        for tokenlist in parse_incr(data_file):
+            for token in tokenlist:
+                error_flag = False
+                if token["lemma"] == "_":  # or token['upos'] in ('PUNCT', 'SYM')
+                    # flag = True
+                    continue
 
-            initial = bool(token["id"] == 1)
-            token_form = token["form"].lower() if initial else token["form"]
+                initial = bool(token["id"] == 1)
+                token_form = token["form"].lower() if initial else token["form"]
 
-            candidate = lemmatizer.lemmatize(token_form, lang=language)
-            greedy_candidate = greedy_lemmatizer.lemmatize(token_form, lang=language)
+                try:
+                    candidate = lemmatizer.lemmatize(token_form, lang=language)
+                except Exception:
+                    candidate = ""
```

Comment:
What kind of exception are you getting?
If you get exceptions we should fix simplemma 🙂 

I noticed that tokens that reach the prefix strategy and the whole token match a prefix, throw.
Have you find any other case?

Reply:
Yes, it's the IndexError I also saw, do you want to fix it?

Reply:
I just made sure the script could run but I'll remove this line as soon as we get it fixed.